lib/hawkular/prometheus/prometheus_api.rb

```diff
@@ -0,0 +1,50 @@
+require 'hawkular/base_client'
+require 'hawkular/inventory/entities'
+require 'ostruct'
+
+module Hawkular::Prometheus
+  class AlerterClient < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      entrypoint = normalize_entrypoint_url entrypoint, 'hawkular/alerter'
+      @entrypoint = entrypoint
+
+      super(entrypoint, credentials, options)
+    end
+
+    def prometheus_entrypoint
+      rest_client('/prometheus/endpoint').get
+    end
+  end
+
+  # Interface to talk with the Prometheus server used for Middleware Manager
+  # @param entrypoint [String] base url of Hawkular Services
+  class Client < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      prometheus_entrypoint = AlerterClient.new(entrypoint, credentials, options).prometheus_entrypoint
+      entrypoint = normalize_entrypoint_url prometheus_entrypoint, 'api/v1'
+      @entrypoint = entrypoint
```

Comment:
`@entrypoint` is set by `BaseClient`, right? I wouldn't also set it here, then.

If not, at least there's no need to set the temporary variable, so:

```ruby
@entrypoint = normalize_entrypoint_url(prometheus_entrypoint, 'api/v1')
```

Reply:
+1